suite-native/navigation/src/components/TabBar.tsx

```diff
@@ -31,6 +35,14 @@ const tabBarStyle = prepareNativeStyle<{
 export const TabBar = ({ state, navigation, tabItemOptions }: TabBarProps) => {
     const { applyStyle } = useNativeStyles();
     const insets = useSafeAreaInsets();
+    const { isUsbDeviceConnectFeatureEnabled } = useIsUsbDeviceConnectFeatureEnabled();
+    const isDeviceImported = useSelector(selectIsSelectedDeviceImported);
+
+    const isPortfolioEmpty = useSelector(selectIsPortfolioEmpty);
+
+    const isConnectedDevice = isUsbDeviceConnectFeatureEnabled && !isDeviceImported;
+
+    if (isPortfolioEmpty && !isConnectedDevice) return null;
```

Comment:
I think this logic should be rather in the `module-home`. Could not we hide the tabBar from there with something like this?  https://stackoverflow.com/a/72658508

Reply:
The `module-home` shouldn't control tab bar, `TabBar.tsx` should. Why do you think it's better?

Reply:
Because you need to import unnecessary packages to the `suite-native/navigation` module. It works in this scenarion, but since the `navigation` module is imported in almost every package we have, it is very inclined to cyclical dependency.  For this reason i think it is bad and non-reusable design.


I still believe the hiding of the tab bar should be handled in the specific screen source file.

Reply:
I will have to solve this also in receive flow, and there would be the cyclical dependency occur. So let's find out some common solution together 👍

Reply:
Cyclical dependency won't be increased. Navigation is now used in almost all modules same as wallet-core. It's designed to sustain this scenario right now. 

I believe it shouldn't be controlled by the function you sent, seems too hacky IMO and this solution is much cleaner and deterministic.

Reply:
I would say it's fine to have it there, from that two added packages is one from common which will never import anything from native and second is feature-flags which should never need use navigation.

Also module-home is not only place that is responsible for TabBar because every item in TabBar is basically module so all of them have same rights to mess with it.

I would say let's keep it there because in the end it's the simplest solution without using any "hacks".

Reply:
OK, but I will still need to find some "hack" for hiding and showing tab bar based on the `@suite-native/receive` module. 

But I don't want to block this issue anymore with this. ✅

Reply:
@PeKne You should be able to track which screens are "present" in navigation package and show tabbar based on that

Reply:
I was looking for something that hides the tabbar when you are already in the screen and the screen state changes. But after discussion with design, It is not needed anymore.